src/dokan/posix_acl.h

```diff
@@ -0,0 +1,166 @@
+#ifndef CEPH_POSIX_ACL_H
```

Comment:
I am very confused by what's happening with the "posix acl" stuff here. It looks like it's independently re-implementing ACL semantics? Isn't there a library we can use so it's known-good?

Reply:
Not sure if it's relevant, but all Andreas' work seems to be lgpl 2.1: http://savannah.nongnu.org/projects/acl

Reply:
Thanks for the suggestion, nice to see that's actually maintained. At a quick glance, I think it would require a few changes in order to build using Mingw but it's definitely worth considering, at least for the next iteration.

Reply:
Even better, looks like we can drop all the permission checks performed at ceph-dokan level and leave that up to libcephfs.

The only edge case that we have is the delete operation, in which the ``Delete`` Dokan callback is supposed to tell if the delete operation is allowed, having the actual delete operation performed in the ``Cleanup`` callback.